### ISSUE TITLE
Possible Cross-Variant functionality

### ISSUE THREAD

**Issue by eliotwrobson** [PrairieLearn/PrairieLearn#6465]
Right now, there's no way to enable any kind of cross-variant functionality with questions/elements. There might be demand for this (for example, with the `pl-hidden-hints` element, see discussion: https://github.com/PrairieLearn/PrairieLearn/pull/6334#discussion_r985847482). This could be useful in some cases, but could be confusing and cause issues if there isn't enough support for content development (for example, some way to manipulate the cross-variant counter in the case of hints). This definitely warrants some discussion before any possible implementation.

**Comment by nwalters512** [comment]
I'm going to go on the record as saying that at this point in time, I think implementing this would be a poor idea. As pointed out here, it would likely be confusing to develop against. More broadly, I think we should work hard to maintain the simple mental model of variants being self-contained units; I don't like the precedent that this would set.

**Comment by mwest1066** [comment]
Also see #26 for a limited subset of this general concept.

**Comment by reteps** [comment]
Steven Bogaerts requested this feature on Slack to avoid the same variant being provided to students multiple times.

**Comment by stevenbogaerts** [comment]
I'm ok with "the same variant being provided multiple times", I'd just like students to see each main variant once before repeating them, for example. Even if there were some way I could generate a list of variants and set a flag somewhere to state "random selection" or "ordered selection", that would be great. It doesn't have to be full communication between variants, just a way to let students see each one. Maybe I just need to make my variants separate questions, but there is something satisfying about putting closely related versions all under the umbrella of one question.

**Comment by nwalters512** [comment]
Note that PrairieLearn has no idea which abstract variants of a question exist. Every time a student needs a new one, we run Python code which randomly creates a new one. This makes it effectively impossible to implement something like "ordered selection" - there's no "set" of variants from which to make selections.

**Comment by stevenbogaerts** [comment]
Makes sense! If I were able to pass, say, a single int into the creation of a variant, I could manage the ordering myself. Let the initial value be 0, and increment it by 1 each time, something like that. Or people could do something more complex with other int values if they wanted.

Alternatively, if there were a counter automatically passed in saying "This is how many variants the current user has seen", that could be useful for my purpose, and would still mostly maintain the "self-contained units" idea of variants.
